model/githubapp/github_app_installation.go

```diff
@@ -323,21 +266,21 @@ func createCacheID(installationID int64, permissions *github.InstallationPermiss
 	// be concatenated into the cache ID.
 	for i := 0; i < permissionsStructVal.NumField(); i++ {
 		field := permissionsStructVal.Field(i)
-		if field.Kind() != reflect.Ptr || field.Elem().Kind() != reflect.String {
+		if field.Kind() != reflect.Ptr ||
+			field.Elem().Kind() != reflect.String ||
+			field.IsNil() {
 			continue
 		}
 
-		if !field.IsNil() {
-			fieldValue, ok := field.Interface().(*string)
-			if !ok {
-				return "", errors.New("failed to convert permission field to string")
-			}
-			permissionPairs = append(permissionPairs, fmt.Sprintf("%s:%s", permissionsStructVal.Type().Field(i).Name, utility.FromStringPtr(fieldValue)))
+		fieldValue, ok := field.Interface().(*string)
+		if !ok {
+			continue
```

Comment:
I changed `!ok` to continue instead of return an error because:
1. It should be impossible- all permissions are strings.
2. In the usage, we always skip on error- but this is the only error, so it reduces down our complexity.

Reply:
If we're going to remove the error, can we log that this happened? If it hits this case, it's a bug and would result in the cache creating an invalid ID invisibly, which could cause a lot of undetected caching issues.